### PR TITLE
Allow multiple minimisers

### PR DESCRIPTION
**Pull request by ltratt**
This PR makes it possible in the future to add other state minimisation approaches than Pager's. First, it makes a tweak to the API. This:

```rust
yacc_to_statetable(grms)
```

becomes this:

```rust
yacc_to_statetable(grms, Minimiser::Pager)
```

Then it splits various files apart so that the stategraph module, in particular, is no longer hard-coded to the Pager way of doing things. Most of this is relatively simple moving, but the diffs end up looking a bit scarier than they actually are.

### CODE REVIEW

File: src/lib/stategraph.rs [hunk 1]
```diff
@@ -58,143 +55,7 @@ use grammar::{PIdx, Grammar, Symbol, SIdx};
...
-
-/// The type of "context" (also known as "lookaheads")
-pub type Ctx = BitVec;
-
-#[derive(Clone, Debug, PartialEq)]
-pub struct Itemset {
-    pub items: HashMap<(PIdx, SIdx), Ctx, BuildHasherDefault<FnvHasher>>
-}
-
 impl Itemset {
```
Comment (ptersilie) [inline src/lib/stategraph.rs]: why is this left over, despite ItemSet being moved into its own module?

Comment (ptersilie) [inline src/lib/stategraph.rs]: ah, is it because `stategraph.rs` will become the pager algorithm, so it adds weakly compatible to the ItemSet module?

Comment (ptersilie) [inline src/lib/stategraph.rs]: Okay, I've now seen the next commit where you move `stategraph.rs` to `pager.rs`, so everything makes sense.

**Comment by ptersilie** [comment]
Looks good. Do you want to rebase this?

**Comment by ltratt** [comment]
I think the commits are good separately, as then each one is a simple logical change. If you're OK with that, then feel free to merge as-is.

**Comment by ptersilie** [comment]
Alright. Merged.

**Pull request merged**